lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -849,7 +851,39 @@ def test_student_progress_with_valid_and_invalid_id(self, default_store):
         # Assert that valid 'student_id' returns 200 status
         self.assertEqual(resp.status_code, 200)
 
-    def test_non_asci_grade_cutoffs(self):
+    def test_unenrolled_student_progress_for_credit_course(self):
+        """
+        Check if in credit course progress check for the student who is not enrolled in course raises None or not
+        """
+        # Create a new course, a user which will not be enrolled in course, admin user for staff access
+        course = CourseFactory.create()
+        not_enrolled_user = UserFactory.create()
+        self.request.user = AdminFactory.create()
+
+        # Create and enable Credit course
+        CreditCourse.objects.create(course_key=course.id, enabled=True)
+
+        # Configure a credit provider for the course
+        CreditProvider.objects.create(
+            provider_id="ASU",
+            enable_integration=True,
+            provider_url="https://credit.example.com/request",
+        )
+
+        requirements = [{
+            "namespace": "grade",
+            "name": "grade",
+            "display_name": "Grade",
+            "criteria": {"min_grade": 0.52},
+        }]
+        # Add a single credit requirement (final grade)
+        set_credit_requirements(course.id, requirements)
+
+        resp = views.progress(self.request, course_id=course.id.to_deprecated_string(),
+                              student_id=not_enrolled_user.id)
+        self.assertEqual(resp.status_code, 200)
+
+    def test_non_ascii_grade_cutoffs(self):
```

Comment:
@Ayub-Khan , you're right, correcting this from `asci` to `ascii` is out of scope of your PR. However, if you see a silly error like this that can really trivially be fixed, it's just worth fixing.